### PR TITLE
[SYNPYTH-7] Docstring

### PR DESCRIPTION
**Pull request by dancio**

### CODE REVIEW

File: docs/source/getting_started.rst [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+and add a :class:`~syncano.models.base.ApiKey` to it::
+
+    >>> import syncano
+    >>> connection = syncano.connect(api_key='abcd')
+    >>> instance = connection.instances.create(name='dummy_test', description='test')
+    >>> instance
+    <Instance: dummy_test>
+
+    >>> api_key = instance.api_keys.create()
+    >>> ApiKey: 47>
```
Comment (atteroTheGreatest) [inline docs/source/getting_started.rst]: Typo?

File: syncano/exceptions.py [hunk 2]
```diff
@@ -18,10 +18,15 @@ def __str__(self):
 
 
 class SyncanoValueError(SyncanoException):
-    pass
+    """An Python :class:`ValueError` error occurred."""
```
Comment (atteroTheGreatest) [inline syncano/exceptions.py]: A Python...

File: syncano/exceptions.py [hunk 3]
```diff
@@ -45,10 +50,14 @@ def __str__(self):
 
 
 class SyncanoValidationError(SyncanoValueError):
-    pass
+    """An validation error occurred."""
```
Comment (atteroTheGreatest) [inline syncano/exceptions.py]: An -> A

File: syncano/exceptions.py [hunk 4]
```diff
@@ -59,4 +68,4 @@ def __str__(self):
 
 
 class SyncanoDoesNotExist(SyncanoException):
-    pass
+    """An doesn't exist error occurred."""
```
Comment (atteroTheGreatest) [inline syncano/exceptions.py]: An what?

File: syncano/models/base.py [hunk 5]
```diff
@@ -63,7 +65,7 @@ def create_error_class(cls):
         )
 
     def build_doc(cls, name, meta):
-        '''Give the class a docstring if it\'s not defined.'''
+        """Give the class a docstring if it\'s not defined."""
```
Comment (atteroTheGreatest) [inline syncano/models/base.py]: You don't have to escape single quote here.

File: syncano/models/manager.py [hunk 6]
```diff
@@ -187,6 +214,7 @@ def list(self, *args, **kwargs):
 
     @clone
     def first(self, *args, **kwargs):
+        """Returns the first object matched by the lookup parameters, or None if there is no matching object."""
```
Comment (atteroTheGreatest) [inline syncano/models/manager.py]: Misplaced comma, comma should be after 'None'.

File: syncano/models/manager.py [hunk 7]
```diff
@@ -274,6 +308,7 @@ def serialize(self, data, model=None):
         return model(**properties) if self._serialize else data
 
     def request(self, method=None, path=None, **request):
+        """Internal method which calls Syncano API and returns serialized data."""
```
Comment (atteroTheGreatest) [inline syncano/models/manager.py]: Comma before which, or change which to 'that' end don't add a comma.

Comment (23doors) [inline syncano/models/manager.py]: It's actually a restrictive clause. There should be no comma, even when using which.

File: syncano/models/manager.py (hunk 6 above)
Comment (23doors) [inline syncano/models/manager.py]: No comma before if unless conjunction is moved to the beginning.

Comment (atteroTheGreatest) [inline syncano/models/manager.py]: Comma wars! ;) You might be right here.

**Pull request merged**